pkg/azurefile/azurefile.go

```diff
@@ -108,28 +115,51 @@ func (d *Driver) Run(endpoint string) {
 	s.Wait()
 }
 
-func (d *Driver) checkFileShareExists(accountName, resourceGroup, name string) (bool, error) {
+func (d *Driver) checkFileShareExists(accountName, resourceGroup, name string, accountKey *string) (bool, error) {
 	// find the access key with this account
-	accountKey, err := d.cloud.GetStorageAccesskey(accountName, resourceGroup)
+	key, err := d.cloud.GetStorageAccesskey(accountName, resourceGroup)
 	if err != nil {
 		// Because the account does not exist in the resource group, the file share could not exist.
 		// In the next step, we will create a temp account to create file share.
 		// So here we return nil, and warn the user.
 		klog.Warningf("could not get storage key for storage account %s: %v", accountName, err)
 		return false, nil
 	}
+	if accountKey != nil {
+		*accountKey = key
+	}
 
-	fileClient, err := d.getFileSvcClient(accountName, accountKey)
+	fileClient, err := d.getFileSvcClient(accountName, key)
 	if err != nil {
-		return false, fmt.Errorf("error creating azure client: %v", err)
+		return false, err
 	}
 	return fileClient.GetShareReference(name).Exists()
 }
 
+func (d *Driver) checkCapacity(accountName, accountKey, fileShareName string, requestGiB int) error {
+	fileClient, err := d.getFileSvcClient(accountName, accountKey)
+	if err != nil {
+		return err
+	}
+	resp, err := fileClient.ListShares(azs.ListSharesParameters{Prefix: fileShareName})
+	if err != nil {
+		return fmt.Errorf("error listing file shares: %v", err)
+	}
+	for _, share := range resp.Shares {
+		// When the requestGiB is null, "CreateShare" will set the quota to max value "5120".
```

Comment:
a little confused about the logic here:
 - `requestGiB` is set by user, so if `requestGiB` is 0, then it should return error, right?